lib/src/main/java/com/auth0/msg/AbstractMessage.java

```diff
@@ -0,0 +1,250 @@
+//        }
+        return null;
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param Key the signing key
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(Key key, Algorithm algorithm) throws InvalidClaimsException, SerializationException {
+        return null;
+    }
+
+    /**
+     * verify that the required claims are present
+     * @return whether the verification passed
+     */
+    protected boolean verify() {
+        //This method will set error if verification fails
+        List<String> errors = new ArrayList<String>();
+
+        List<String> reqClaims = getRequiredClaims();
+        if (!reqClaims.isEmpty() && this.claims.isEmpty()){
+            errors.add("The required claims are missing");
+            return false;
+        }
+
+        StringBuilder errorSB = new StringBuilder();
+        for (String req: reqClaims) {
+            if (!claims.containsKey(req)) {
+                errors.add("This message is missing required claim: " + req);
+                errorSB.append(req);
+            }
+        }
+
+        if (errorSB.length() != 0) {
+            errors.add("Message is missing required claims:" + errorSB.toString());
+            return false;
+        }
+
+        List<String> customClaims = new ArrayList<String>();
+        for(String claimName : claims.keySet()) {
+            // if knownClaim, validate claim
+            if (ClaimsValidator.isKnownClaim(claimName)) {
+                Boolean valid = ClaimsValidator.validate(claimName, claims.get(claimName), getMessageType());
+                if (!valid) {
+                    errors.add(claimName + "is an invalid claim");
+                }
+            } else {
+                customClaims.add(claimName);
+            }
+        }
+        if (!errors.isEmpty()) {
+            String aggregateError = "";
+            for (String err: errors){
+                aggregateError += err;
+            }
+            throw new InvalidClaimsException(aggregateError);
+        }
+        return false;
+    }
+
+    public abstract MessageType getMessageType();
+
+    /**
+     * add the claim to this instance of message
+     * @param String the name of the claim
+     * @param Object the value of the claim to add to this instance of Message
+     * @return a Message representation of the Json
+     */
+    public void addString(String name, Object value) {
+        // verify 'name’ is a valid claim and then check the type is valid before adding
+    }
+
+    /**
+     * @return Error an object representing the error status of claims verification
+     */
+    public Error getError() {
+        return error;
+    }
+
+    /**
+     * @return List of the list of claims for this messsage
+     */
+    public Map<String, Object> getClaims(){
+//        verify();
+        return this.claims;
```

Comment:
This should handle the logic -

if(!verrified){
verify();
}
if(error != null){
throw new invalidclaimsexception();
}
return this.claims;